samples/navigate-route-with-rerouting/src/main/java/com/esri/arcgismaps/sample/navigateroutewithrerouting/components/NavigateRouteWithReroutingViewModel.kt

```diff
@@ -0,0 +1,400 @@
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.navigateroutewithrerouting.components
+
+import android.app.Application
+import android.speech.tts.TextToSpeech
+import android.text.format.DateUtils
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.Geometry
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.location.LocationDisplayAutoPanMode
+import com.arcgismaps.location.RouteTrackerLocationDataSource
+import com.arcgismaps.location.SimulatedLocationDataSource
+import com.arcgismaps.location.SimulationParameters
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.LocationDisplay
+import com.arcgismaps.navigation.DestinationStatus
+import com.arcgismaps.navigation.ReroutingParameters
+import com.arcgismaps.navigation.ReroutingStrategy
+import com.arcgismaps.navigation.RouteTracker
+import com.arcgismaps.navigation.TrackingStatus
+import com.arcgismaps.tasks.networkanalysis.RouteParameters
+import com.arcgismaps.tasks.networkanalysis.RouteResult
+import com.arcgismaps.tasks.networkanalysis.RouteTask
+import com.arcgismaps.tasks.networkanalysis.Stop
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.navigateroutewithrerouting.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.launch
+import java.io.File
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicBoolean
+
+class NavigateRouteWithReroutingViewModel(application: Application) :
+    AndroidViewModel(application) {
+
+    // Path of the San Diego transport network used by the route task
+    private val provisionPath: String by lazy {
+        application.getExternalFilesDir(null)?.path.toString() +
+                File.separator + application.getString(R.string.navigate_route_with_rerouting_app_name)
+    }
+
+    // Passed to the composable MapView to set the mapViewProxy
+    val mapViewProxy = MapViewProxy()
+
+    // The map to display basemap, route traveled & route ahead graphics
+    val map = ArcGISMap(BasemapStyle.ArcGISStreets)
```

Comment:
```suggestion
    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISStreets)
```

Just because map in Kotlin can refer to hash maps or mapping values from one to another